### ISSUE TITLE
`localorie#load_translations` fails when using Nvim with Packer.nvim

### ISSUE THREAD

**Issue by JonasPhilbert** [airblade/vim-localorie#7]
I've used NeoVim over vim for quite some time now, but only recently changed my `.vimrc` out for an `init.lua` configuration, using only Lua to configure NeoVim.

With the change, I switched from using https://github.com/junegunn/vim-plug to using https://github.com/wbthomason/packer.nvim for package management. This has seemingly broken localorie.

When writing to the translation file `config/locales/en.yml`, I get the following error:
```
"config/locales/en.yml" 6189L, 238913B written
Error detected while processing function localorie#load_translations[2]..<SNR>149_reject[8]..<SNR>149_reject[8]..<SNR>149_reject[8]..<SNR>149_reject:
line    4:
E735: Can only compare Dictionary with Dictionary
E735: Can only compare Dictionary with Dictionary
```

Also:
- Calling `localorie#expand_key()` echoes nothing.
- `localorie#translate()` works sporadically.

It is very likely that I've just missed something, but I'm at a loss as to why this occurs.

**Comment by airblade** [comment]
I can’t imagine why changing package manager would make any difference, nor for that matter whether you write your config in Vimscript or Lua.

Has your translations file changed?  Does it have a “file” key by any chance?

**Comment by JonasPhilbert** [comment]
Me neither. I am probably missing something.

I've checked, and there are keys in the translation file that are called `file`, but not in the outermost scope. Always buried within other keys.

Turns out the `localorie#expand_key()` issue was due to my own oversight. Thus, it does seem to work now, even though the error when writing the translation file persists.

I'll just close this for now, seen as I can work with it. Thanks for your consideration.

> I can’t imagine why changing package manager would make any difference, nor for that matter whether you write your config in Vimscript or Lua. Has your translations file changed? Does it have a “file” key by any chance?
> […](#)
> On 12 Aug 2022, at 11:46, Jonas Philbert ***@***.***> wrote: ﻿ I've used NeoVim over vim for quite some time now, but only recently changed my .vimrc out for an init.lua configuration, using only Lua to configure NeoVim. With the change, I switched from using https://github.com/junegunn/vim-plug to using https://github.com/wbthomason/packer.nvim for package management. This has seemingly broken localorie. When writing to the translation file config/locales/en.yml, I get the following error: "config/locales/en.yml" 6189L, 238913B written Error detected while processing function localorie#load_translations[2]..<SNR>149_reject[8]..<SNR>149_reject[8]..<SNR>149_reject[8]..<SNR>149_reject: line 4: E735: Can only compare Dictionary with Dictionary E735: Can only compare Dictionary with Dictionary Also: Calling localorie#expand_key() echoes nothing. localorie#translate() works sporadically. It is very likely that I've just missed something, but I'm at a loss as to why this occurs. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.

**Comment by airblade** [comment]
> I've checked, and there are keys in the translation file that are called file, but not in the outermost scope. Always buried within other keys.

I think this is triggering a bug in the plugin.  I'm away at the moment so can't easily verify it, but I'll look into it properly when I'm back.

**Comment by airblade** [comment]
Yes, there was a bug.  It's fixed in 9d4c7fc.

Thanks for reporting this!
